### PR TITLE
Make isXXX available for bool properties only

### PR DESCRIPTION
**Pull request by kesselb**
I'm not sure about this one. Probably not necessary but without isXXX is just an alias for getXXX. I fail somehow to revert the whitespace changes :confused:

### CODE REVIEW

File: lib/public/AppFramework/Db/Entity.php [hunk 1]
```diff
@@ -156,6 +156,17 @@ public function __call($methodName, $args) {
...
 	}
 
+	/**
+	 * @param string $methodName
+	 * @return bool
+	 */
+	protected function isGetterForBoolProperty(string $methodName): bool {
+		if (strpos($methodName, 'is') === 0) {
+			$fieldName = lcfirst(substr($methodName, 2));
+			return isset($this->_fieldTypes[$fieldName]) && $this->_fieldTypes[$fieldName] === 'bool';
```
Comment (ChristophWurst) [inline lib/public/AppFramework/Db/Entity.php]:
```suggestion
			return $this->_fieldTypes[lcfirst(substr($methodName, 2))]) ?? 'int' === 'bool';
```

**Review by ChristophWurst — APPROVED** [review]
Looks good!

These whitespace changes are fine :)

File: lib/public/AppFramework/Db/Entity.php (hunk 1 above)
Comment (kesselb) [reply lib/public/AppFramework/Db/Entity.php]: Fine by me but very hard to read ;)

Comment (ChristophWurst) [reply lib/public/AppFramework/Db/Entity.php]: Fair enough :) I'm fine with either

File: tests/lib/AppFramework/Db/EntityTest.php [hunk 2]
```diff
@@ -226,5 +230,18 @@ public function testFieldsNotMarkedUpdatedIfNothingChanges() {
...
+		$this->assertIsBool($entity->isTrueOrFalse());
+	}
 
+	/**
+	 * @expectedException BadFunctionCallException
+	 */
+	public function testIsGetterShoudFailForOtherType() {
+		$entity = new TestEntity();
+		$entity->setName('hello');
+		$this->assertIsBool($entity->isHello());
```
Comment (kesselb) [inline tests/lib/AppFramework/Db/EntityTest.php]: :rofl:

File: lib/public/AppFramework/Db/Entity.php (hunk 1 above)
Comment (kesselb) [reply lib/public/AppFramework/Db/Entity.php]: I was about to change it to your suggestion ;) But we need another check. settype accepts `bool` and `boolean`.

https://github.com/nextcloud/mail/blob/18a76774eb0f0e00b5a3ddfbea8d2e980219b642/lib/Db/Mailbox.php#L71
https://github.com/nextcloud/richdocuments/blob/9b3fb584fb1c9fb7ce506a0797f061be07320833/lib/Db/Wopi.php#L111

Comment (kesselb) [reply lib/public/AppFramework/Db/Entity.php]: `strpos($this->_fieldTypes[lcfirst(substr($methodName, 2))] ?? 'notype', 'bool') === 0;`

Perhaps something like this. But this is even more obfuscated.

Comment (ChristophWurst) [reply lib/public/AppFramework/Db/Entity.php]: :see_no_evil: no, then let's use the multi-line approach :)

**Comment by rullzer** [comment]
So... good to go?

**Review by ChristophWurst — APPROVED** [review]
Yes

**Review by rullzer — APPROVED** [review]

**Pull request merged**